app/models/role.rb

```diff
@@ -1,10 +1,16 @@
 class Role < ActiveRecord::Base
   belongs_to :resource, polymorphic: true
   has_and_belongs_to_many :users, join_table: :users_roles
-
+  before_destroy :cancel
```

Comment:
If we are just returning false, we can do oneliner `before_destroy { false }`. Also, can we add please add comment about why we are doing this?